ReactiveObjCBridge/ObjectiveCBridging.swift

```diff
@@ -101,119 +112,200 @@ extension QueueScheduler {
-	return Result<(), NSError>.error(message, file: file, line: line)
-}
+extension RACScheduler {
+	/// Create a `RACScheduler` that wraps the given scheduler.
+	///
+	/// - parameters:
+	///   - scheduler: The `Scheduler` to wrap.
+	///
+	/// - returns: A `RACScheduler` that schedules blocks to `scheduler`.
+	public convenience init(_ scheduler: Scheduler) {
+		self.init(racSwiftScheduler: RACSwiftScheduler(wrapping: scheduler))
+	}
 
-/// Create a `SignalProducer` which will subscribe to the provided signal once
-/// for each invocation of `start()`.
-///
-/// - parameters:
-///   - signal: The signal to bridge to a signal producer.
-///   - file: Current file name.
-///   - line: Current line in file.
-///
-/// - returns: Signal producer created from the provided signal.
-public func bridgedSignalProducer<Value>(from signal: RACSignal<Value>, file: String = #file, line: Int = #line) -> SignalProducer<Value?, AnyError> {
-	return _bridgedSignalProducer(from: signal, file: file, line: line)
+	/// Create a `RACScheduler` that wraps the given scheduler.
+	///
+	/// - parameters:
+	///   - scheduler: The `DateScheduler` to wrap.
+	///
+	/// - returns: A `RACScheduler` that schedules blocks to `scheduler`.
+	public convenience init(_ scheduler: DateScheduler) {
+		self.init(racSwiftScheduler: RACSwiftScheduler(wrapping: scheduler))
+	}
 }
 
-private func _bridgedSignalProducer<Value>(from signal: RACSignal<Value>, file: String = #file, line: Int = #line) -> SignalProducer<Value?, AnyError> {
-	return SignalProducer<Value?, AnyError> { observer, disposable in
-		let next: (_ value: Value?) -> Void = { obj in
-			observer.send(value: obj)
+private final class RACSwiftScheduler: RACScheduler {
+	enum Backing {
+		case scheduler(Scheduler)
+		case dateScheduler(DateScheduler)
+	}
+
+	private let base: Backing
+
+	init(wrapping base: Scheduler) {
+		self.base = .scheduler(base)
+	}
+
+	init(wrapping base: DateScheduler) {
+		self.base = .dateScheduler(base)
+	}
+
+	private func wrap(_ block: @escaping () -> Void) -> () -> Void {
+		return {
+			Thread.current.threadDictionary["RACSchedulerCurrentSchedulerKey"] = self
+			block()
+			Thread.current.threadDictionary["RACSchedulerCurrentSchedulerKey"] = nil
 		}
+	}
+
+	open override func schedule(_ block: @escaping () -> Void) -> RACDisposable? {
+		switch base {
+		case let .scheduler(scheduler):
+			return scheduler.schedule(wrap(block)).map(RACDisposable.init)
 
-		let failed: (_ error: Swift.Error?) -> Void = { error in
-			observer.send(error: AnyError(error ?? defaultNSError("Nil RACSignal error", file: file, line: line)))
+		case let .dateScheduler(scheduler):
+			return scheduler.schedule(wrap(block)).map(RACDisposable.init)
 		}
+	}
+
+	open override func after(_ date: Date, schedule block: @escaping () -> Swift.Void) -> RACDisposable? {
+		switch base {
+		case let .scheduler(scheduler):
+			Thread.sleep(until: date)
```

Comment:
`RACScheduler` says that [this method should block with an immediate scheduler](https://github.com/ReactiveCocoa/ReactiveObjC/blob/master/ReactiveObjC/RACScheduler.h#L79). In this case, while it isn't really an immediate scheduler, it is treated as one given the lack of time-based scheduling.